src/ShellClients/PanelWindow.vala

```diff
@@ -30,22 +52,34 @@ public class Gala.PanelWindow : Object {
             }
         });
 
-        window.stick ();
-
-        clone = new PanelClone (wm, this);
-
-        unowned var display = wm.get_display ();
-
-        window_positioner = new WindowPositioner (display, window, WindowPositioner.Position.from_anchor (anchor));
-
-        notify["anchor"].connect (() => window_positioner.position = WindowPositioner.Position.from_anchor (anchor));
+        notify["anchor"].connect (() => position = Position.from_anchor (anchor));
 
-        unowned var workspace_manager = display.get_workspace_manager ();
+        unowned var workspace_manager = window.display.get_workspace_manager ();
         workspace_manager.workspace_added.connect (update_strut);
         workspace_manager.workspace_removed.connect (update_strut);
 
         window.size_changed.connect (update_strut);
         window.position_changed.connect (update_strut);
+
+        default_gesture_tracker = new GestureTracker (ANIMATION_DURATION, ANIMATION_DURATION);
+
+        notify["hidden"].connect (() => {
```

Comment:
what is `hidden`? I can't find it anywhere in the code

Reply:
Ah right that's a copy paste remnant (the notify issue :sweat_smile:)

While looking whether we still need that I came across an issue in the hidetracker where if hide is triggered again while one was already scheduled the old hide id would be overwritten which in turn caused for it to not be reset by a call to show while it was still queued. It's a bit tricky to reproduce but can be seen if you have the dock set to autohide on overlap and move a window over it, move it while it overlaps the dock and then move it away from the dock all in the time before it hides. This would cause the dock to stay hidden even though it should show.
It happens in main as well but it's fixed now here :)